open_schools_platform/organization_management/circles/tests/views/test_circle_invite_teacher.py

```diff
@@ -0,0 +1,122 @@
+
+    def test_teacher_can_change_status_on_accepted(self):
+        circle = create_test_circle()
+        user = create_logged_in_user(self)
+        teacher_profile = user.teacher_profile
+        teacher = create_teacher(teacher_profile=teacher_profile, circle=circle, name='test_teacher')
+        query = create_test_query_circle_invite_teacher(circle, teacher, teacher_profile)
+        data = {
+            "id": query.id,
+            "status": Query.Status.ACCEPTED
+        }
+        response = self.client.patch(self.query_status_change_url, data)
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(Query.Status.ACCEPTED, get_query(filters={"id": str(query.id)}).status)
+
+    def test_teacher_can_change_status_on_declined(self):
+        circle = create_test_circle()
+        user = create_logged_in_user(self)
+        teacher_profile = user.teacher_profile
+        teacher = create_teacher(teacher_profile=teacher_profile, circle=circle, name='test_teacher')
+        query = create_test_query_circle_invite_teacher(circle, teacher, teacher_profile)
+        data = {
+            "id": query.id,
+            "status": Query.Status.DECLINED
+        }
+        response = self.client.patch(self.query_status_change_url, data)
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(Query.Status.DECLINED, get_query(filters={"id": str(query.id)}).status)
+
+    def test_circle_can_change_status_on_canceled(self):
+        circle_owner = create_logged_in_user(self)
+        circle = create_test_circle_with_user_in_org(user=circle_owner)
+        user = create_test_user(phone="+79998880008")
+        teacher_profile = user.teacher_profile
+        teacher = create_teacher(teacher_profile=teacher_profile, circle=circle, name='test_teacher')
+        query = create_test_query_circle_invite_teacher(circle, teacher, teacher_profile)
+        data = {
+            "id": query.id,
+            "status": Query.Status.CANCELED
+        }
+        response = self.client.patch(self.query_status_change_url, data)
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(Query.Status.CANCELED, get_query(filters={"id": str(query.id)}).status)
+
+    def test_circle_cannot_change_status_not_to_canceled(self):
+        circle_owner = create_logged_in_user(self)
+        circle = create_test_circle_with_user_in_org(circle_owner)
+        user = create_test_user(phone="+79998880008")
+        teacher_profile = user.teacher_profile
+        teacher = create_teacher(teacher_profile=teacher_profile, circle=circle, name='test_teacher')
+        query = create_test_query_circle_invite_teacher(circle, teacher, teacher_profile)
+        data = {
+            "id": query.id,
+            "status": Query.Status.ACCEPTED
+        }
+        response = self.client.patch(self.query_status_change_url, data)
+        self.assertNotEqual(200, response.status_code)
```

Comment:
I think it is better to assert it with status code, that is raised by exceptions returned in our views.